source/includes/fundamentals/code-snippets/MonolightCodec.java

```diff
@@ -0,0 +1,57 @@
+package fundamentals.monolightcodec;
+
+import org.bson.BsonReader;
+import org.bson.BsonType;
+import org.bson.BsonWriter;
+import org.bson.codecs.Codec;
+import org.bson.codecs.DecoderContext;
+import org.bson.codecs.EncoderContext;
+import org.bson.codecs.configuration.CodecRegistry;
+
+// start class
+public class MonolightCodec implements Codec<Monolight>{
+
+    private Codec<String> powerStatusCodec;
+    private Codec<Integer> integerCodec;
+
+    public MonolightCodec(CodecRegistry registry) {
+        this.integerCodec = registry.get(Integer.class);
+        this.powerStatusCodec = registry.get(String.class);
+    }
+
+    @Override
+    public void encode(BsonWriter writer, Monolight value, EncoderContext encoderContext) {
+        writer.writeStartDocument();
+        writer.writeName("powerStatus");
+        powerStatusCodec.encode(writer, value.getPowerStatus(), encoderContext);
+        writer.writeName("colorTemperature");
+        writer.writeInt32(value.getColorTemperature());
+        writer.writeEndDocument();
+    }
+
+    @Override
+    public Monolight decode(BsonReader reader, DecoderContext decoderContext) {
+        Monolight monolight = new Monolight();
+
+        reader.readStartDocument();
+        while (reader.readBsonType() != BsonType.END_OF_DOCUMENT) {
+            String fieldName = reader.readName();
+            if (fieldName.equals("powerStatus")) {
+                monolight.setPowerStatus(powerStatusCodec.decode(reader, decoderContext));
```

Comment:
I think although we have a `powerStatusCodec` this codec would handle its own encoding for such a simple task. But I understand that it could be a more complex type.

Reply:
If this doesn't seem necessary, I can rework the example to exclude the PowerStatusCodec. My intent was to build up the example by referencing the PowerStatusCodec presented in an earlier section, and demonstrate how you might use it to decode a field but it could be too contrived/simple. Let me know what you think.

Reply:
I think to fix there are two choices for a quick fix: 

1) Keep PowerStatusCodec, just don't register it in the global codec registry and instantiate it here.
2) Change the type for the field to an Enum, that way there will be no global registry clash.